web/packages/teleterm/src/ui/services/clusters/clustersService.ts

```diff
@@ -395,6 +395,13 @@ export class ClustersService extends ImmutableStore<types.ClustersServiceState>
     }
   }
 
+  async removeAppGateway(appUri: uri.AppUri) {
+    const gateway = this.findGatewayByConnectionParams(appUri, '');
+    if (gateway) {
+      await this.removeGateway(gateway.uri);
+    }
+  }
+
```

Comment:
I don't think we need `removeAppGateway`. `disconnect` can just do what it does for the db gateways which is take the gateway URI and delete it.

Kube gateways have this special case only because `getConnectionGatewayKubeOperations` needs to support both the old kube connections (which don't have `gatewayUri` and an underlying gateway) and new ones (which do have a gateway).

Could you add a comment above the call to `removeAppGateway` in `getConnectionGatewayKubeOperations` about it? It took me a sec to remember why kube connections are special. We could also add a `TODO` in `removeAppGateway` to remove this method and use `removeGateway` after getting rid of the old kube document implementation.